autonomx/ComputeEngine.cpp

```diff
@@ -78,8 +79,238 @@ void ComputeEngine::receiveOscData(int id, QVariantList data) {
+    QList<QVariant> dataAsList = data;
+    int inputValue = 0;
+
+    // check if generator exists -> if not, ignore
+    if(!generatorsHashMap->contains(generatorId)) return;
+
+    // split input osc control message into pieces
+    QStringList controlMessageList = controlMessage.split(QLatin1Char('/'));
+    if(controlMessageList.length() < 2) return;
+
+    // assign split osc message pieces to corresponding variables
+    for(int i = 0; i < controlMessageList.length(); i++) {
+        switch(i) {
+        case 0:
+            generatorName = controlMessageList.at(0).toLower();
+            break;
+        case 1:
+            inputType = controlMessageList.at(1).toLower();
+            break;
+        case 2:
+            parameter1 = controlMessageList.at(2).toLower();
+            break;
+        case 3:
+            parameter2 = controlMessageList.at(3).toLower();
+            break;
+        }
+    }
+
+    // if the first parameter is empty, message was entered incorrectly
+    if(parameter1 == "") {
+        qDebug() << "empty control message entered!";
+        return;
+    }
+
+    // check if targetted generator name exists -> else, ignore
+    QSharedPointer<Generator> generator = generatorsHashMap->value(generatorId);
+    if(!(generatorName == generator->getGeneratorName().toLower())) {
+        // name not found -> skip and return
+        return;
+    }
+
+    // verify dataList isn't empty
+    if(!dataAsList.isEmpty()) {
+        if(dataAsList.length() == 1) {
+            inputValue = dataAsList.at(0).toInt();
+        }
+    }
+
+    if(parameter1.contains("_") || parameter2.contains("_")) {
+        parameter1.remove(QChar('_'));
+        parameter2.remove(QChar('_'));
+    }
+
+    // if user-input control message better-matched our QProperty system formatting -> many types of messages could be handled here
+//    if(parameterControlList.contains(parameter1)) {
+//        if(inputValue < 1) inputValue = 1;
+
+//        QByteArray controlMessageArray = parameter1.toLocal8Bit();
+//        const char *controlMessageArrayChar = controlMessageArray.data();
+//        generator->setProperty(controlMessageArrayChar, inputValue);
+//        return;
+//    }
+
+    // check if control message is global or generator-specific
+    if(parameterControlList.value(parameter1) == "global") {
+        if(inputValue < 1) inputValue = 1;
+
+        // control message is global parameter
+        if(parameter1 == "width") {
+            generator->writeLatticeWidth(inputValue);
+        } else if(parameter1 == "height") {
+            generator->writeLatticeHeight(inputValue);
+        } else if(parameter1 == "speed") {
+            generator->writeSpeed(inputValue);
+        } else if(parameter1 == "restart") {
+            generator->initialize();
+        } else if(parameter1 == "reset") {
+            generator->resetParameters();
+        } else if(parameter1 == "reset_regions") {
```

Comment:
We could match the name of the methods more closely. "resetRegions", not "reset_regions".

Reply:
Can the parameter1 be reset_regions if the underscores are removed?